design/admin/templates/setup/datatype_code.tpl

```diff
@@ -26,7 +26,7 @@ class {$full_class_name} extends eZDataType
     */
     function {$full_class_name}()
     {literal}{{/literal}
-        $this->eZDataType( self::{$constant_name}, "{$desc_name}" );
+        parent::__construct( self::{$constant_name}, "{$desc_name}" );
```

Comment:
Didn't we choose keep some of these legacy constructors for core classes for compatibility with wide array of extensions available out there?

Reply:
We did: https://github.com/ezsystems/ezpublish-legacy/blob/master/kernel/classes/ezdatatype.php#L47-L89

But this is new code and to avoid deprecation warnings this is the right answer if you'd be creating an extension for newer versions of legacy.

Reply:
Ah, okay, no worries then! :+1: